### ISSUE TITLE
Spike: PEFT (LoRA) fine-tune of qwen2.5-coder:7b to clear the QE test-gen oracle floor

### ISSUE THREAD

**Issue by proffesor-for-testing** [proffesor-for-testing/agentic-qe#550]
## Hypothesis

A **PEFT (LoRA/MicroLoRA) fine-tune of `qwen2.5-coder:7b`** can lift its QE test-gen oracle score from ~0.50 toward the `qwen3-coder:30b` band (~0.62) — at a fraction of the 30B's latency/footprint. If it clears the QE quality floor, a fine-tuned 7B becomes a viable free-tier default, strengthening the local-model story without the 30B cost.

## Motivating data (committed in `value-score.ts`, 2026-06-29)

Small-coder oracle measurements (mutation-graded, durable-first test-gen):
- `qwen2.5-coder:7b` = **0.50** (4/8) — already beats the general 8B at 0.38
- `qwen2.5-coder:1.5b` = 0.08
- Reference band: `qwen3:30b-a3b` = 0.62, `qwen3-coder:30b` (new free-tier default), `claude-sonnet-4-6` = 0.83

The 7B sits just below the 0.6 mutation threshold — the curve that motivates a targeted fine-tune rather than just scaling up. Related: D3 model-floor finding (qwen3:8b below floor, 30b clears it) and the Darwin+ruvllm plan (MicroLoRA/SONA does the "weight half" around a frozen cheap model).

## Existing harness (reuse, don't rebuild)

`scripts/oracle-model-bench-local.ts` already grades local Ollama models with the mutation oracle and reports gen tok/s + kill rate:
```
tsx scripts/oracle-model-bench-local.ts qwen2.5-coder:7b
ATTEMPTS=2 tsx scripts/oracle-model-bench-local.ts qwen2.5-coder:7b   # best-of-2
```
This is the eval that produces the comparable score; the fine-tuned model just needs to be registered/served so the same harness can grade it.

## Plan / scope

1. **Baseline (reproduce):** re-run the 7B benchmark on the multi-fixture set, n≥30 fixtures for low-noise, record the per-fixture distribution (not just the aggregate).
2. **Training data:** curate test-gen trajectories where the mutation oracle scored high (durable-first tests that killed mutants) — the oracle is the label source, so this is self-supervised from our own arena/oracle runs.
3. **PEFT run:** LoRA/MicroLoRA on `qwen2.5-coder:7b` via the ruvllm/SONA path (host Ollama at `host.docker.internal:11434`); keep the base frozen.
4. **Eval:** grade the adapted model with `oracle-model-bench-local.ts` on a **held-out** fixture set (train/eval disjoint — see #549 darwin-guard contamination guard).
5. **Decision:** does fine-tuned-7B clear the 0.6 floor and approach 0.62? Report score, tok/s, and a cost/quality point vs 30B and cloud.

## Acceptance criteria

- [ ] Reproduced 7B baseline with per-fixture distribution (n≥30), committed alongside the existing value-score numbers.
- [ ] PEFT adapter trained on oracle-labeled test-gen data, base model frozen.
- [ ] Held-out eval (train/eval disjoint, asserted) graded by the existing mutation oracle harness.
- [ ] Reported: adapted-7B oracle score, gen tok/s, and a cost/quality comparison vs `qwen3-coder:30b` and `claude-sonnet-4-6`.
- [ ] Go/no-go: promote fine-tuned-7B as a free-tier option only if it clears the floor on the held-out set.

## Notes / risks

- Reproduction-first: trust the mutation-oracle number on a real fixture set, not a single function (the harness already uses multi-fixture).
- Keep any training data / adapter handling within the unified-memory + ADR constraints; no second store-of-record.
- This is the "weight half" of the Darwin+ruvllm self-learning plan — pairs with the harness-evolution (Darwin) half.
